patches/chrome-browser-ssl-https_upgrades_interceptor.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/chrome/browser/ssl/https_upgrades_interceptor.cc b/chrome/browser/ssl/https_upgrades_interceptor.cc
+index 4e7b8c18e5049f0c20148259ee5755789e089ad4..81875827cf4f672efe71c17b1ade82c706429740 100644
+--- a/chrome/browser/ssl/https_upgrades_interceptor.cc
++++ b/chrome/browser/ssl/https_upgrades_interceptor.cc
+@@ -237,6 +237,7 @@ bool ShouldExcludeNavigation(content::NavigationUIData* navigation_ui_data,
+     level = NavigationRequestSecurityLevel::kCaptivePortalLogin;
+   }
+ #endif
++  BRAVE_SHOULD_EXCLUDE_NAVIGATION
```

Comment:
Is there a test for this? I'm not sure that it's doing the same thing as the original, but the original code doesn't have any comments explaining what it is doing and neither does the new code (which should be changed)

Reply:
also if this is trying to do what I think then I'm not even sure this is the right place to do this because it's supposed to be controlled through `force_no_https_upgrade` which could be done in `BraveContentBrowserClient::WillCreateURLLoaderRequestInterceptors`

Reply:
I believe the [original change](https://github.com/brave/brave-core/pull/19822/files) was to get around [this upstream's limitation](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ssl/https_upgrades_interceptor.cc;l=329;drc=0a1b1b32a6141b2e8816af54734a1628997d4794;bpv=1;bpt=0):
```
// If the URL was typed with an explicit http:// URL, it is opted-out from
// upgrades.
```

The new upstream change moved the check for URLs typed explicitly with HTTP to a function that also checks if the URL is a captive portal. On our side we modify the function to preserve upstream's behavior in regards to captive portals, but continue with our original change to allow typed HTTP URLs upgrades.

@arthuredelstein's original change includes browser tests for typed URLs, so we should still be covered by those. Upstream added tests for captive portals: `HttpsUpgradesBrowserTest.CaptivePortal*`, but we are currently disabling all upstream's `HttpsUpgradesBrowserTest.*` tests. Maybe we can the portal tests as ours in a follow-up?

I'll add a comment for this change.

Reply:
Added a comment in https://github.com/brave/brave-core/pull/24227/commits/f7b7ff19d174a8395717384cf4a5ec22892e0e91.
Also, turned back on a bunch of upstream's `HttpsUpgradesBrowserTest` tests, including the `CaptivePortal*` tests in https://github.com/brave/brave-core/pull/24227/commits/d77d20c53413038b8e66f0aa1674a947721997c4

Reply:
Done in 0818b25fef16a9365f4926e71df0a5d8b9787bda